linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -585,6 +585,14 @@ final class Emitter private (config: CommonPhaseConfig,
     }
 
     if (classEmitter.needInstanceTests(linkedClass)) {
+      if (!linkedClass.hasInstances && kind.isClass) {
+        /* The isInstanceOf implementation will generate
+         * `x instanceof $c_TheClass`, but `$c_TheClass` won't be declared at
+         * all. Define it as a fake class to avoid `ReferenceError`s.
+         */
+        addToMainBase(classEmitter.genFakeClass(linkedClass))
+      }
+
```

Comment:
Hmmm... This makes me wonder:

- If we put this into `GlobalKnowledge` could we simply emit `false` instead?
- Do we still need the normal instance tests (at least the `is` part), if this is a class?

Reply:
OK scratch that. I found a relatively elegant way to do it. Re-updated.

Reply:
> We could, but I think that is a job for the optimizer, not for the emitter.

Fair enough.